File: JULY-2025/31-07-2025/p898.swift
class Solution {
    func subarrayBitwiseORs(_ arr: [Int]) -> Int {
        let n = arr.count
        var res: [Int] = []
        
        for i in 0..<n{
            var or = 0
            for j in i..<n{
                or = or | arr[j]
                if or != res.last{
                    res.append(or)
                }
            }
        }
        
        return Set(res).count
    }
}

File: JULY-2025/31-07-2025/p898.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int SubarrayBitwiseORs(int[] arr) {
        HashSet<int> result = new HashSet<int>();
        HashSet<int> current = new HashSet<int>();
        
        foreach (int num in arr) {
            HashSet<int> next = new HashSet<int> { num };
            foreach (int prev in current) {
                next.Add(prev | num); // OR with previous results
            }
            current = next;
            foreach (int val in current) {
                result.Add(val); // store unique results
            }
        }

        return result.Count;
    }
}